cardstack/boxel
# Add payment setup flow upon registration

Refer to https://linear.app/cardstack/issue/CS-7396/send-user-to-stripe-checkout-for-free-plan-during-registration-flow

### What is changing
- Add payment setup flow right after the user signs up and verifies it successfully.
- If the user skips this step, prompt them to set up a payment method until it's completed.

Note:
- I have tweaked the overall basic font size (16px to 13px) as per design when translating into the app. I found that if we apply the font size to 16px, users will need to scroll more when they have a smaller screen size, e.g., a 13-inch MacBook.
- There is a discussion to apply this similar flow after the user registered but has not subscribed yet (https://github.com/cardstack/boxel/pull/1756#discussion_r1832966855), which is included in this PR.
- Account popover is shown at payment setup, so user able see their account and option to logout

### How to test 
There is some pre-setup needed to enable free plan on development account:

1. Use stripe cli to listen for the webhooks that Stripe sends to the realm server

```
stripe listen --forward-to localhost:4201/_stripe-webhook --api-key sk_test_api_key_from_the_sandbox_account
```

2. You will get webhook signing secret from stripe cli after Step 1 is done

```
> Ready! You are using Stripe API Version [x]. Your webhook signing secret is whsec_xxxxx
```

3. Go to `packages/realm-server`, pass STRIPE_WEBHOOK_SECRET & STRIPE_API_KEY environment value and start the realm server

```
STRIPE_WEBHOOK_SECRET=... STRIPE_API_KEY=... pnpm start:all
```

4. Go to `packages/realm-server` and run the following command to sync the stripe products to the database, make sure you have the stripe api key set

```
STRIPE_API_KEY=... pnpm sync-stripe-products
```

5. Perform "Setup up Secure Payment Method" flow. Subscribe with valid test card [here](https://docs.stripe.com/testing#cards)

You should be able to subscribe successfully after you perform the steps above.


### Screenshots
**Video**

https://github.com/user-attachments/assets/9317790b-0b2c-4b42-9b8a-8b5ec6d35513

**Monitor size**

![Screenshot 2024-11-21 at 5 36 37 PM](https://github.com/user-attachments/assets/ab3750bb-4373-4043-82ed-eb53bdb39fec)

**Laptop Size**

<img width="1424" alt="Screenshot 2024-11-21 at 5 38 17 PM" src="https://github.com/user-attachments/assets/55f4d93f-3fee-4bfb-becb-fdcc1e94776c">
